### ISSUE TITLE
Implement: Add pausing flags to `DelegationManager.undelegate` and `DelegationManager.forceUndelegation`

### ISSUE THREAD

**Issue by ChaoticWalrus** [Layr-Labs/eigenlayer-contracts#165]
## Description
We should have pausing flags for these functions, but they aren't implemented yet.

### Action Items
- [x] Implement changes to the contracts on a separate branch
- [x] Update tests and scripts as necessary
- [x] Update documentation to reflect changes
- [ ] Get PR reviewed
- [ ] Respond to review / make any further changes
- [ ] Create ticket for tests
- [ ] Merge PR

### Blocking Issues
Is anything blocking for this? (Do we need to design and/or build something first?)
N/A

**Comment by ChaoticWalrus** [comment]
Functions have been revised as part of a PR https://github.com/Layr-Labs/eigenlayer-contracts/pull/176 , which unblocked this.
Still need to add the pausing flag to `DelegationManager.undelegate` (i.e. get this ticket itself done) -- planning to do this in a separate, small PR.

**Comment by ChaoticWalrus** [comment]
PR is open for actually getting this done, and has been linked (https://github.com/Layr-Labs/eigenlayer-contracts/pull/193) -- this issue should be closed once the PR is merged, as it also includes unit test coverage.
Noting here for context that the `DelegationManager.forceUndelegation` function was removed as part of the previous PR mentioned above -- it was merged into `DelegationManager.undelegate`, so we only need the new modifier on a single function.
